File: src/min_product/min_product.py
n = int(input())
numbers = list(map(int, input().split()))

numbers.sort()
res_1 = numbers[0] * numbers[1]
res_2 = numbers[0] * numbers[-1]
res_3 = numbers[-1] * numbers[-2]

print(min(res_1, res_2, res_3))

File: src/06_min_product/min_product.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main(void) {
    int n = 0;
    cin >> n;
    vector<int> v(n);
    for (int i = 0; i < n; i++) {
        cin >> v[i];
    }
    sort(v.begin(), v.end());
    long int res1 = v[0] * v[1];
    long int res2 = v[n - 1] * v[n - 2];
    long int res3 = v[0] * v[n - 1];
    cout << min(res1, min(res2, res3)) << endl;
    return 0;
}